### PR TITLE
Codebase Quality: Network drives service and view model

### PR DESCRIPTION
**Pull request by lukeblevins**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Verify network storage devices show up correctly

**Screenshots (optional)**
N/A

### CODE REVIEW

File: src/Files.Backend/Services/INetworkDrivesService.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Files.Backend.Services
+{
+	public interface INetworkDrivesService
+	{
+		/// <summary>
+		/// Enumerates network storage devices
+		/// </summary>
+		/// <returns>A collection of network storage devices</returns>
+		IAsyncEnumerable<ILocatableFolder> GetDrivesAsync();
```
Comment (yaira2) [inline src/Files.Backend/Services/INetworkDrivesService.cs:16]: Thank you for adding the docs

**Review by yaira2 — APPROVED** [review]
LGTM

File: src/Files.App/ServicesImplementation/NetworkDrivesService.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+using Files.Backend.Services;
+using Files.Sdk.Storage.LocatableStorage;
+using Vanara.PInvoke;
+using Vanara.Windows.Shell;
+
+namespace Files.App.ServicesImplementation
+{
+	public class NetworkDrivesService : INetworkDrivesService
+	{
+		public bool DisconnectNetworkDrive(ILocatableFolder drive)
```
Comment (d2dyno1) [inline src/Files.App/ServicesImplementation/NetworkDrivesService.cs:15]: Small nit, but not that important: I'd be nice if you could add "<inheritdoc />" to signify that a method is documented 🙂

**Pull request merged**